### PR TITLE
test: use `T.Setenv` to set env vars in tests

### PR DESCRIPTION
**Pull request by Juneezee**
### What does this PR do?

This PR replaces `os.Setenv` with `t.Setenv`. Starting from Go 1.17, we can use `t.Setenv` to set environment variable in test. The environment variable is automatically restored to its original value when the test and all its subtests complete. This ensures that each test does not start with leftover environment variables from previous completed tests.

```go
func TestFoo(t *testing.T) {
	// before
	key := "ENV"
	originalEnv := os.Getenv(key)
	
	if err := os.Setenv(key, "new value"); err != nil {
		t.Fatal(err)
	}
	defer func() {
		if err := os.Setenv(key, originalEnv); err != nil {
			t.Logf("failed to set env %s back to original value: %v", key, err)
		}
	}()
	
	// after
	t.Setenv(key, "new value")
}
```

### Motivation

Testing cleanup. This saves us at least 2 lines (error check, and unsetting the env var) on every instance.

### Describe how to test/QA your changes

All tests should continue to pass.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

File: pkg/runtime/runtime_test.go [hunk 1]
```diff
@@ -21,17 +20,17 @@ func TestAutoMaxProcs(t *testing.T) {
...
 	SetMaxProcs()
 	assert.Equal(t, 1, runtime.GOMAXPROCS(0))
 
-	os.Setenv("GOMAXPROCS", "1500m")
+	t.Setenv("GOMAXPROCS", "1500m")
 	// set new limit
 	SetMaxProcs()
 	assert.Equal(t, 1, runtime.GOMAXPROCS(0))
 
-	os.Setenv("GOMAXPROCS", "2000m")
```
Comment (djmitche) [inline pkg/runtime/runtime_test.go:34]: Did you verify that this correctly handles multiple calls for the same variable?  The docs don't say anything about it, but if you imagine it's calling `t.Cleanup` internally, then it's probably OK as cleanups run in the reverse of the order in which `t.Cleanup` was called.

**Review by djmitche — APPROVED** [review]
It'd be good to add something about this under https://github.com/DataDog/datadog-agent/tree/main/docs/dev.  Maybe `agent_tests.md` can be expanded to include some advice about unit tests?

File: pkg/runtime/runtime_test.go (hunk 1 above)
Comment (brycekahle) [reply pkg/runtime/runtime_test.go:34]: From docs:
> Setenv calls os.Setenv(key, value) and uses Cleanup to restore the environment variable to its original value after the test.

Comment (djmitche) [reply pkg/runtime/runtime_test.go:34]: I guess you don't have to imagine.  So, yes, that seems to imply that it's safe to use multiple times in a test.

Comment (Juneezee) [reply pkg/runtime/runtime_test.go]: > Did you verify that this correctly handles multiple calls for the same variable? The docs don't say anything about it, but if you imagine it's calling `t.Cleanup` internally, then it's probably OK as cleanups run in the reverse of the order in which `t.Cleanup` was called.

@djmitche I've updated `TestAutoMaxProcs` into table test. This should makes it clearer :smiley:

File: pkg/process/config/config_test.go [hunk 2]
```diff
@@ -263,11 +257,11 @@ func TestDefaultConfig(t *testing.T) {
...
-	os.Setenv("DOCKER_DD_AGENT", "yes")
+	t.Setenv("DOCKER_DD_AGENT", "yes")
 	_ = NewDefaultAgentConfig()
 	assert.Equal(os.Getenv("HOST_PROC"), "")
 	assert.Equal(os.Getenv("HOST_SYS"), "")
-	os.Setenv("DOCKER_DD_AGENT", "no")
+	t.Setenv("DOCKER_DD_AGENT", "no")
 	assert.Equal(config.DefaultProcessExpVarPort, config.Datadog.GetInt("process_config.expvar_port"))
 
 	os.Unsetenv("DOCKER_DD_AGENT")
```
Comment (robertjli) [inline pkg/process/config/config_test.go:267]: Remove `Unsetenv()` here too

**Review by robertjli — CHANGES_REQUESTED** [review]

**Review by robertjli — APPROVED** [review]
LGTM for Processes. Thank you for this change!

**Review by juliogreff — APPROVED** [review]

**Review by knusbaum — APPROVED** [review]

**Pull request merged**